e2e/testsuite/grpc_query.go

```diff
@@ -20,6 +28,63 @@ import (
 	ibcexported "github.com/cosmos/ibc-go/v7/modules/core/exported"
 )
 
+// GRPCClients holds a reference to any GRPC clients that are needed by the tests.
+// These should typically be used for query clients only. If we need to make changes, we should
+// use E2ETestSuite.BroadcastMessages to broadcast transactions instead.
+type GRPCClients struct {
+	ClientQueryClient     clienttypes.QueryClient
+	ConnectionQueryClient connectiontypes.QueryClient
+	ChannelQueryClient    channeltypes.QueryClient
+	FeeQueryClient        feetypes.QueryClient
+	ICAQueryClient        controllertypes.QueryClient
+	InterTxQueryClient    intertxtypes.QueryClient
+
+	// SDK query clients
+	GovQueryClient    govtypesv1beta1.QueryClient
+	GovQueryClientV1  govtypesv1.QueryClient
+	GroupsQueryClient grouptypes.QueryClient
+	ParamsQueryClient paramsproposaltypes.QueryClient
+	AuthQueryClient   authtypes.QueryClient
+	AuthZQueryClient  authz.QueryClient
+
+	ConsensusServiceClient tmservice.ServiceClient
+}
+
+// InitGRPCClients establishes GRPC clients with the given chain.
+// The created GRPCClients can be retrieved with GetChainGRCPClients.
+func (s *E2ETestSuite) InitGRPCClients(chain *cosmos.CosmosChain) {
```

Comment:
should [`GetChainGRPCClients`](https://github.com/cosmos/ibc-go/blob/47d678eb57d35a410f9a51baf47aa7a0d110df2c/e2e/testsuite/testsuite.go#L396) also be moved in this file?

Reply:
I think this makes sense, the one thing I see is that not all of the clients are query clients, there is one `ConsensusServiceClient` that is being initialised here too.

I wonder would it make sense to have two separate functions init-ing query/service clients and then calling then `InitGRPCClients` calling both.